pkg/mflag/flag.go

```diff
@@ -1058,7 +1058,7 @@ func (fs *FlagSet) Parse(arguments []string) error {
 		case ContinueOnError:
 			return err
 		case ExitOnError:
-			os.Exit(2)
+			os.Exit(125)
```

Comment:
working on feedback now, I've fixed the 'image doesn't not exist' errors by adding a similar check in api/client/run.go for the already existing derr.ErrorCodeNoSuchImageHash and derr.ErrorCodeNoSuchImageTag errors that were created in daemon/create.go --should have seen this before :(   working on other feedback now....